### PR TITLE
Feature Admin view of user accounts

### PR DESCRIPTION
**Pull request by neelxie**
#### What does this PR do
Create a page for admin to view all user accounts

#### Screenshots
image

### CODE REVIEW

**Comment by neelxie** [comment]
> @neelxie This is good.
> I'm guessing you had to do this without input from the design end... Which is interesting.
> Have you considered adding a few columns for more info e.g. `The date a user account was created`??

@StephenAraka let me add that.

File: src/components/UserAccounts/index.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  isRetrieving: false,
+  isFetched: false,
+};
+
+export const mapStateToProps = (state) => {
+  const { isRetrieving, users, isFetched } = state.UsersListReducer;
+  return { isRetrieving, users, isFetched };
+};
+
+export const mapDispatchToProps = (dispatch) => bindActionCreators({
```
Comment (Mubangizi) [inline src/components/UserAccounts/index.js:111]: Please remove `bindActionCreators`

Comment (neelxie) [reply src/components/UserAccounts/index.js:111]: @Mubangizi on it

File: src/components/UserAccounts/index.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+// assigning defaults
+UserAccounts.defaultProps = {
+  users: [],
+  isRetrieving: false,
+  isFetched: false,
+};
+
+export const mapStateToProps = (state) => {
+  const { isRetrieving, users, isFetched } = state.UsersListReducer;
+  return { isRetrieving, users, isFetched };
```
Comment (Mubangizi) [inline src/components/UserAccounts/index.js:108]: Also for uniformity, when you use `isFetched`, use `isFetching` and when you use `isRetrieved` and use `isRetrieving` for the props values, this creates conflicts and errors now like on the projects page, it uses `isRetrieving` to get projects and the same page tries to use props for `UsersListReducer`, meaning `isRetrieving` for users will update `isRetrieving` for projects. Does this make sense?

File: src/redux/reducers/usersListReducer.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import {
+  GET_USERS_SUCCESS,
+  GET_USERS_FAIL,
+  START_GETTING_USERS,
+} from '../actions/actionTypes';
+
+const initialState = {
+  users: [],
+  isFetched: false,
+  isFetching: false,
```
Comment (Mubangizi) [inline src/redux/reducers/usersListReducer.js:10]: Actually You are using is Fetching here.

**Comment by Mubangizi** [comment]
But There is Already `actions` and `reducers` for users list on `develop`, you could rebase and remove any conflicts.

File: src/components/UserAccounts/index.js (hunk 1 above)
Comment (neelxie) [reply src/components/UserAccounts/index.js]: @Mubangizi done

File: src/components/UserAccounts/index.js (hunk 2 above)
Comment (neelxie) [reply src/components/UserAccounts/index.js:108]: @Mubangizi cool

File: src/redux/reducers/usersListReducer.js (hunk 3 above)
Comment (neelxie) [reply src/redux/reducers/usersListReducer.js:10]: @Mubangizi this too has been fixed

**Comment by neelxie** [comment]
> But There is Already `actions` and `reducers` for users list on `develop`, you could rebase and remove any conflicts.

@Mubangizi thank you let me do that

**Comment by neelxie** [comment]
> > But There is Already `actions` and `reducers` for users list on `develop`, you could rebase and remove any conflicts.
> 
> @Mubangizi thank you let me do that

@Mubangizi rebase is done too.

**Comment by Mubangizi** [comment]
Sorry this is the rebase i was talking about, this is the one that has conflicts

**Comment by Mubangizi** [comment]
Hey about `email` addresses and `roles` look and feel, this looks neater to me like when they are interchanged, what do you say?
Screenshot from 2020-04-15 15-30-30

**Comment by MutegekiHenry** [comment]
> Hey about `email` addresses and `roles` look and feel, this looks neater to me like when they are interchanged, what do you say?
> Screenshot from 2020-04-15 15-30-30

Yeah i like this arrangement better too!

**Comment by neelxie** [comment]
> > Hey about `email` addresses and `roles` look and feel, this looks neater to me like when they are interchanged, what do you say?
> > Screenshot from 2020-04-15 15-30-30
> 
> Yeah i like this arrangement better too!

@Mubangizi good suggestion I have worked on this.
image

**Pull request merged**